stdlib/public/runtime/Float16Support.cpp

```diff
@@ -150,6 +151,14 @@ SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
   return __gnu_f2h_ieee(f);
 }
 
+/// Convert from Float16 to long double.  Since Float32 covers the entire range
+/// of Float16 values and since we already know how to convert Float32 to long
+/// double (which, at least on x86, doesn't involve function calls), we just
+/// let the compiler do the latter part for us.
+SWIFT_RUNTIME_EXPORT long double __extendhfxf2(short h) {
+  return __gnu_h2f_ieee(h);
+}
```

Comment:
We should add this conditionally I think.  FP80 is not supported on Windows, Android, and Apple platforms.

```suggestion
#if (defined(__i386__) || defined(__x86_64__)) && !(defined(__ANDROID__) || defined(__APPLE__) || defined(_WIN32))

SWIFT_RUNTIME_EXPORT long double __extendhfxf2(short h) {
  return __gnu_h2f_ieee(h);
}

#endif
```

Reply:
Do we think conditionalising it is really necessary? `long double` compiles everywhere these days, no? (On platforms without native support for anything larger, it tends to turn into `double`, possibly stored in a larger type, I think? On PowerPC I think it would have been that weird thing with two `double` values in it.)

Reply:
This clearly does the right thing for any "long double", so there's no harm in compiling it on any platform, although because it has `xf` in the name, it should only be called on platforms where `long double` is in fact Intel 80-bit extended format.

Reply:
Yes, but it does add to the runtime (and standard library size).  Plus, it is an extra interface that now becomes part of the ABI of the library.

Reply:
Additionally (and I hadn't paid enough attention to this), there is already a huge conditional wrapping the entire file, which will need adjusting slightly as some of these functions will be needed for Linux.